SO-4809: Make inclusion of non-property type changes optional during ...

AAAlinaaa opened the pull request:
... compare, exclude them from branch compare results, count total
commit details in new compare result content

AAAlinaaa committed:
SO-4809: Make inclusion of non-property type changes optional during ...

... compare, exclude them from branch compare results, count total
commit details in new compare result content

cmark commented on commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java:
```diff
@@ -142,27 +142,27 @@ public RevisionCompare compare(final String branch) {
 	@Override
-	public RevisionCompare compare(final String baseBranch, final String compareBranch, final int limit) {
-		return compare(getBranchRef(baseBranch), getBranchRef(compareBranch), limit);
+	public RevisionCompare compare(final String baseBranch, final String compareBranch, final int limit, boolean preserveComponentChanges) {
```
I think a better configuration option would be the opposite, `excludeComponentChanges`. The default functionality is to always include them, so allowing optional exclusion of them is the functionality we need. Please rename.

cmark replied:
Alternatively, one could define the set of Operation literal to exclude.

cmark commented on commons/com.b2international.index/src/com/b2international/index/revision/RevisionCompare.java:
```diff
@@ -138,14 +144,16 @@ public RevisionCompare build() {
 	private final int totalAdded;
 	private final int totalChanged;
 	private final int totalRemoved;
+	private final int total;
```
What is total here? Total `added` + `changed` + `removed`?

AAAlinaaa replied:
No, it is the total, possibly more than the sum of the three if it is more than the limit.

AAAlinaaa committed:
SO-4809: Ensure compare result tallies are not restricted to the ...

... applicable limit, exclude containers as changed components from
branch compare results

AAAlinaaa committed:
SO-4809: Change configuration option from preserve to exclude ...

...component changes

cmark commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/BranchCompareResult.java:
```diff
@@ -79,7 +79,7 @@ public Builder totalDeleted(int totalDeleted) {
 		}
-		
+				
 		public BranchCompareResult build() {
```
I think you can safely remove the `build(Set changedContainers)` method and keep the zero-arg build method as the one that actually builds the result.

AAAlinaaa committed:
SO-4809: Default to unparameterized branch compare result builder

cmark reviewed (approved):
LGTM!
